source/common/router/router.h

```diff
@@ -264,6 +274,7 @@ class Filter : Logger::Loggable<Logger::Id::router>,
   UpstreamRequestPtr upstream_request_;
   Http::HeaderMap* downstream_headers_{};
   Http::HeaderMap* downstream_trailers_{};
+  std::list<std::tuple<std::string, std::string, int>> downstream_set_cookies_;
```

Comment:
I think it might be a good idea to create an explicit type instead of using tuple here.
Why `std::list` instead of `std::vector`?